test/src/testPVCopy.cpp

```diff
@@ -289,6 +289,7 @@ static void testMasterField(PVRecordPtr const& pvRecord)
         cout << "Master PV structure from copy" << endl << *pvMasterField << endl;
         cout << "Master PV structure from copy offset " << pvMasterField->getFieldOffset() << endl;
     }
+    testOk1(pvCopy->isMasterFieldRequested());
```

Comment:
This only appears to test that PVCopy detects when the master field is requested, showing the changes to PVCopy and pvRecord. How about adding tests for the behavior fix described in the intro to this PR, which appears to have been implemented in `MonitorLocal::dataPut()`?

Reply:
New set of tests have been added that verify correct behavior of the changed set.